packages/eas-cli/src/commands/workflow/runs.ts

```diff
@@ -0,0 +1,135 @@
+import { Flags } from '@oclif/core';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import { EasJsonOnlyFlag } from '../../commandUtils/flags';
+import { getLimitFlagWithCustomValues } from '../../commandUtils/pagination';
+import { AppWorkflowRunsFragment, WorkflowRunStatus } from '../../graphql/generated';
+import { AppQuery } from '../../graphql/queries/AppQuery';
+import Log from '../../log';
+import formatFields from '../../utils/formatFields';
+
+export type WorkflowRunResult = {
+  id: string;
+  status: string;
+  gitCommitMessage?: string | null;
+  gitCommitHash?: string | null;
+  startedAt: string | null;
+  finishedAt: string | null;
+  workflowId: string;
+  workflowName: string | null;
+  workflowFileName: string | null;
+};
+
+function processWorkflowRuns(
+  runs: AppWorkflowRunsFragment['runs'],
+  params: {
+    workflowId?: string | undefined;
+    workflowFileName?: string | undefined;
+    status?: string | undefined;
+  }
+): WorkflowRunResult[] {
+  const { workflowId, workflowFileName, status } = params;
+  return runs.edges
+    .filter(edge => {
+      if (workflowId && edge.node.workflow.id !== workflowId) {
+        return false;
+      }
+      if (workflowFileName && edge.node.workflow.fileName !== workflowFileName) {
+        return false;
+      }
+      if (status && edge.node.status !== status) {
+        return false;
+      }
```

Comment:
I think we probably want these to take place server-side so that the limit is accurate? (add the filters to `workflowRunsPaginated` query).

I'm thinking it might make sense to also add a query to get a workflow by file name, in which case we could fetch the workflow and then make `workflowRunsPaginated` only take a `workflowId` and `status` (so there's no need for `fileName` filter).

So, concretely:
1. Add new query to `WorkflowQuery`, `byAppIdAndFileName(appId: ID!, fileName: String!): Workflow!`
2. Update `workflowRunsPaginated` `WorkflowRunFilterInput` to take `workflowId` and `status`.

Reply:
@wschurman to really do what you suggest here would require universe changes, I believe (I didn't see support for these filters in the resolvers there).

We can do (1) above with the existing schema, by fetching all workflows for the app and finding the right ID for the filename. However, I don't see any support for (2) taking a status value, unless we actually modify universe.

For now, I'll make the changes to at least limit the query correctly if the filename is passed in.

Reply:
Yep, I think the universe changes are a prerequisite for this PR. I believe all our lists filtering is done server side.

Reply:
Well, I think we can still merge this PR, provided we remove the status filtering. We can add that later once universe supports it.

Reply:
I think we'd need to remove the `workflowName` filtering as well then. For example:
- I have two workflows, A and B
- The last 10 runs are A, followed by 5 B runs.
- If I ran `eas workflow:runs --workflowName B --limit 10` it'd return empty

(I think, if I'm reading this correctly)

Reply:
@wschurman I am now using different queries when getting runs by workflow filename, to avoid client side filtering on filename. I am still filtering client side on status for now.

This should be safe to merge, and once we add the correct server side filters in universe, we can rework these commands to use them.